midterm-2023-11-07/code/5040-dara-poposka/script.ts

```diff
@@ -0,0 +1,116 @@
+interface Movie{
+    id: number;
+    title: string;
+    year: number;
+    director: string;
+    genre: string[];
+    plot: string;
+    cast: {
+      actor: string;
+      character: string;
+    }[];
+    oscars: { [oscarType: string]: string };
+    rating: number;
+  }
+  type MovieSorter = (first: Movie, second: Movie) => number;
+  
+  document.addEventListener("DOMContentLoaded", siteCode);
+  
+  let movies: Movie[] = [];
+  
+  async function siteCode() {
+    const data = await loadData();
+    movies = data;
+  
+    displayMovies(movies);
+  
+    const titleSort = document.getElementById("sort-title")!;
+    titleSort.addEventListener("click", sortByTitle);
+  
+    const yearSort = document.getElementById("sort-year")!;
+    yearSort.addEventListener("click", sortByYear);
+  
+    const genreFilter = document.getElementById("genre-filter") as HTMLSelectElement;
+    genreFilter.addEventListener("change", applyGenreFilter);
+  }
+  
+  const titleSorter: MovieSorter = (first, second) => first.title.localeCompare(second.title);
+  const yearSorter: MovieSorter = (first, second) => first.year - second.year;
+  
+  const sortByTitle = () => {
+    const sortedMovies = movies.slice().sort(titleSorter);
+    displayMovies(sortedMovies);
+  };
+  
+  const sortByYear = () => {
+    const sortedMovies = movies.slice().sort(yearSorter);
+    displayMovies(sortedMovies);
+  };
+  
+  const applyGenreFilter = () => {
```

Comment:
Very nice method, with one glaring flaw - it doesn't work. However, correct format and logic.
The problem is that the hardcoded value of the dropdown, does not match the data value (`action` vs. `Action`)
+10